spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -594,4 +710,30 @@ contract RolloverVault is
             token.safeApprove(spender, type(uint256).max);
         }
     }
+
+    /// @dev Computes the underlying amount to be returned after deducting the melding fee.
+    ///      The meld fee is applied based on linear function based on the bond's time remaining to maturity wrt is duration.
+    function _computeMeldFee(IBondController bond, uint256 underlyingAmt) private view returns (uint256) {
+        uint256 maturityTimestampSec = bond.maturityDate();
+        uint256 bondDurationSec = maturityTimestampSec - bond.creationDate();
+        uint256 secondsToMaturity = (
+            maturityTimestampSec > block.timestamp ? maturityTimestampSec - block.timestamp : 0
+        );
+        uint256 appliedFeePerc = MAX_MELD_FEE_PERC.mulDiv(secondsToMaturity, bondDurationSec);
+        return underlyingAmt.mulDiv(appliedFeePerc, HUNDRED_PERC);
+    }
+
+    /// @dev Checks if at least one of the bond's tranches is part of the deployed set.
+    ///      Based on this condition we can infer that the bond is NOT malicious,
+    ///      and accept other sibling tranches from this bond.
+    function _isDeployedBond(IBondController bond, BondTranches memory bt) private view returns (bool) {
```

Comment:
Wondering if this deserves to be its own function.

It's assumed, but not checked in this context, that bt corresponds to bond. The function above this knows it, since it makes the call itself.

We could add an error check into each loop iteration, to make this function more "component like", but thinking since it's only called in one place, we can just fold this into the caller function and not worry about it.